dashboard/app/entities.go

```diff
@@ -514,3 +517,93 @@ func timeDate(t time.Time) int {
 	year, month, day := t.Date()
 	return year*10000 + int(month)*100 + day
 }
+
+type RecipientType int
+
+const (
+	To RecipientType = iota
+	Cc
+)
+
+// t.String() must not be changed as the data is stored in this format.
+func (t RecipientType) String() string {
+	return [...]string{"To", "Cc"}[t]
+}
+
+type RecipientInfo struct {
+	Address mail.Address
+	Type    RecipientType
+}
+
+type Recipients []RecipientInfo
+
+func (r Recipients) Len() int           { return len(r) }
+func (r Recipients) Less(i, j int) bool { return r[i].Address.Address < r[j].Address.Address }
+func (r Recipients) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+
+type RecipientString string
+
+func (r Recipients) String() RecipientString {
+	recStrSlice := RecipientStrSlice(r.StringSlice())
+	return RecipientString(strings.Join(recStrSlice, "|"))
+}
+
+func RecipientStrSlice(emails []RecipientString) []string {
+	var slice []string
+	for _, user := range emails {
+		slice = append(slice, string(user))
+	}
+	return slice
+}
+
+func (r Recipients) StringSlice() []RecipientString {
+	var slice []RecipientString
+	for _, addr := range r {
+		newAddress := strings.ReplaceAll(addr.Address.String(), "|", "\t")
+		slice = append(slice, RecipientString(newAddress+"\n"+addr.Type.String()))
+	}
+	return slice
+}
+
+func SplitEmails(emails []string) (to, cc []RecipientString) {
+	for _, e := range emails {
+		user := strings.Split(e, "\n")
+		if len(user) == 1 || user[1] == "To" {
+			to = append(to, RecipientString(strings.ReplaceAll(user[0], "\t", "|")))
+		} else {
+			cc = append(cc, RecipientString(strings.ReplaceAll(user[0], "\t", "|")))
+		}
+	}
+	return to, cc
+}
+
+// TODO change output to Recipients and use Recipients instead of
+// Maintainers/Cc on BugNotification and BugReport.
+func GetEmails(emails []RecipientString) (to, cc []string) {
```

Comment:
I don't think there should be application logic operating directly on the "recipient" strings stored in the database like this.  Instead, translate them to `RecipientInfo` first.